runtime/gc_ctrl.c

```diff
@@ -520,11 +523,13 @@ CAMLprim value caml_gc_set(value v)
 CAMLprim value caml_gc_minor(value v)
 {
   CAML_INSTR_SETUP (tmr, "");
+  caml_ev_begin(EV_EXPLICIT_GC_MINOR);
   CAMLassert (v == Val_unit);
   caml_request_minor_gc ();
   // call the gc and call finalisers
   caml_process_pending_actions();
   CAML_INSTR_TIME (tmr, "explicit/gc_minor");
+  caml_ev_end(EV_EXPLICIT_GC_MINOR);
```

Comment:
This is not well bracketed if `caml_process_pending_action` raises. You can use the `_exn` variant to end the event and re-raise.
```c
{
  value exn;
  ...
  exn = caml_process_pending_actions_exn();
  CAML_INSTR_TIME (tmr, "explicit/gc_minor");
  caml_ev_end(EV_EXPLICIT_GC_MINOR); 
  caml_raise_if_exception(exn);
  return Val_unit;
}
```